src/lexer/lexer.rs

```diff
@@ -53,19 +60,100 @@ impl Lexer {
     pub fn peek_token(&mut self) -> Result<Token> {
         let current = self.current;
         let nesting = self.nesting;
+        let fstring_stack = self.fstring_stack.clone();
+        let start_of_line = self.start_of_line;
+        let inside_fstring_bracket = self.inside_fstring_bracket;
         let token = self.next_token();
         self.current = current;
         self.nesting = nesting;
+        self.fstring_stack = fstring_stack;
+        self.start_of_line = start_of_line;
+        self.inside_fstring_bracket = inside_fstring_bracket;
         token
     }
```

Comment:
I'm a bit confused about `peek_token`. why does it mutate `self`, Maybe we should call it something other than `peek`?

Reply:
Peek token gets the next token but it saves the current state to rewind it after getting the token. This makes it possible for the parser to know what token is going to come after the current token without actually consuming that token.

This is useful for the parser to decide which rule is going to match the current expression.

I think the peek means looking which implies that this is just looking at the next token not consuming it